### DISCUSSION TITLE
Employ coding guidelines

### DISCUSSION THREAD

**Discussion by auermich93** [iBotPeaches/Apktool#2532] [Ideas]
I thought it would be nice to follow some coding guidelines, primarily in terms of documentation and code structure.
The code base lacks here some points, for instance:

- There is not really any sort of java doc. I would suggest java doc for at least public methods and inline comments for key parts.
- There are a couple of unused methods. Either remove them or declare them as unused.
- There are multiple methods that declare exceptions that are actually not thrown.
- There are classes, e.g. `ApkDecoder` where class variable/constant definitions are scattered throughout the class.

I would like to hear from you.

**Comment by iBotPeaches** [comment]
The javadoc I'm taking more seriously after rebooting the maven libraries. I'm noticing that is quite painful without them on public methods, which is leading to taking a look at the methods exposed or not.

Unused methods I'm aware of and probably just need to clean up.

The exception thrown I ran into during javadoc generation for Maven and I started cleaning up slowly in https://github.com/iBotPeaches/Apktool/pull/2530

The variable/constant scatter - I'm waiting to see if a good linter exists in the Java world that perhaps follows closely with AOSP, so we can enforce these and do a big batch organization.

**Reply by iBotPeaches** [reply]
Took a pass at removing declared, but unused exceptions - https://github.com/iBotPeaches/Apktool/commit/f4c2f9bc5083f89ccdfd32a9b94e28798376cc7b

**Reply by auermich93** [reply]
Great. How about using https://checkstyle.sourceforge.io/?
